src/include/network/services/host.rb

```diff
@@ -175,36 +171,30 @@ def HostsMainDialog(standalone)
       end
 
       UI.ChangeWidget(Id(:table), :Items, table_items)
-      UI.SetFocus(Id(:table)) if Ops.greater_than(Builtins.size(table_items), 0)
+      UI.SetFocus(Id(:table)) if !table_items.empty?
 
-      ret = nil
       loop do
-        UI.ChangeWidget(
-          Id(:edit),
-          :Enabled,
-          Ops.greater_than(Builtins.size(table_items), 0)
-        )
-        UI.ChangeWidget(
-          Id(:delete),
-          :Enabled,
-          Ops.greater_than(Builtins.size(table_items), 0)
-        )
+        UI.ChangeWidget(Id(:edit), :Enabled, !table_items.empty?)
+        UI.ChangeWidget(Id(:delete), :Enabled, !table_items.empty?)
 
         ret = UI.UserInput
-        Builtins.y2debug("ret=%1", ret)
 
         # abort?
         if ret == :abort || ret == :cancel
           ReallyAbortCond(Host.GetModified) ? break : next
         # add host
         elsif ret == :add
+          max = table_items.size
           item = HostDialog(max, term(:empty))
+
           next if item.nil?
-          table_items = Builtins.add(table_items, item)
+
+          table_items = table_items.push(item)
```

Comment:
This (re)assignation is not needed ;) `Array#push` directly appends the item to the array.